src/main/java/org/folio/consortia/service/impl/SharingInstanceServiceImpl.java

```diff
@@ -118,6 +128,48 @@ public SharingInstanceCollection getSharingInstances(UUID consortiumId, UUID ins
     return result;
   }
 
+  @Override
+  @Transactional
+  public void completePromotingLocalInstance(String eventPayload) {
+    log.debug("completePromotingLocalInstance:: parameters eventPayload: {}", eventPayload);
+    try {
+      var promotingEvent = objectMapper.readValue(eventPayload, SharingInstance.class);
+
+      String centralTenantId = tenantService.getCentralTenantId();
+      String sourceTenantId = promotingEvent.getSourceTenantId();
+      String targetTenantId = promotingEvent.getTargetTenantId();
+      checkTenantsExistAndContainCentralTenantOrThrow(sourceTenantId, targetTenantId);
+
+      if (ObjectUtils.notEqual(centralTenantId, targetTenantId)) {
+        log.warn("completePromotingLocalInstance:: promotion failed as targetTenantId: {} does not equal to centralTenantId: {}", targetTenantId, centralTenantId);
```

Comment:
In this case we also should update sharing_instance table, because status will be IN_PROGRESS forever without this change

Reply:
Improved